### PR TITLE
cleanup emu.rs a bit

### PR DESCRIPTION
**Pull request by brandonros**

### CODE REVIEW

File: src/emu.rs [hunk 1]
```diff
@@ -3146,60 +3146,54 @@ impl Emu {
...
-            let sz = ins.len();
-            self.out.clear();
-            formatter.format(&ins, &mut self.out); 
-
-            self.emulate_instruction(&ins, sz, true); 
-
-            if self.force_reload {
-                self.force_reload = false;
+        // get first instruction from iterator
+        let ins = decoder.iter().next().unwrap();
```
Comment (brandonros) [inline src/emu.rs:3175]: this is the real change other than the whitespace/reindent

**Comment by sha0coder** [comment]
the editor that I use normally is NeoVim, but sometimes used vscode also (with tab9).
nice work, probably i modified also things on emu.rs and there are some conflicts i have to check them.

regarding this:
```rust
                if self.cfg.is_64bits {
                    self.regs.rip += sz as u64;
                } else {
                    self.regs.set_eip(self.regs.get_eip() + sz as u64);
                }
                
simplified to:
                self.regs.set_eip(self.regs.get_eip() + sz as u64);
```

I have memory dumps that are PE and are mapped in high addresses, in 64bits we should update rip instead eip for those cases, It could be the case that the mapped address is >32bits this depends on PE headers.
set_eip() clears bits upper to 32bits.

**Comment by brandonros** [comment]
@sha0coder fixed merge conflicts

**Comment by sha0coder** [comment]
to clear all the registers can be done with:      self.regs.clear::<64>();
then i randomize then but you can comment this line:      self.regs.rand();

**Comment by sha0coder** [comment]
in theory the branch is ok now, but is not necesary to clear all the registers just use regs.clear::<64>()

**Pull request merged**